kafka/proxy.go

```diff
@@ -131,19 +144,26 @@ func (h *produceRequestHandler) Handle(requestKeyVersion *kafkaprotocol.RequestK
 	return shouldReply, nil
 }
 
-func (h *produceRequestHandler) extractMessages(req protocol.ProduceRequest) []Message {
+func (h *produceRequestHandler) extractMessages(req sarama.ProduceRequest) []Message {
 	var msgs []Message
+
 	for topic, r := range req.Records {
 		for _, s := range r {
 			if s.RecordBatch != nil {
 				for _, r := range s.RecordBatch.Records {
+					// Fixing indirection here
+					headers := make([]sarama.RecordHeader, len(r.Headers))
+					for i := 0; i < len(r.Headers); i++ {
+						headers[i] = *r.Headers[i]
+					}
```

Comment:
Why do you have to copy array if the type when copied matches?

Reply:
They are different types. `[]sarama.RecordHeader` VS `[]*sarama.RecordHeader` 🤷

Reply:
Ahh ok, pointers :P

Reply:
![](https://media2.giphy.com/media/Y9PqYpY5gRVN6/giphy.gif?cid=790b7611f35c8403482e0f24630ac98041c3134f71103075&rid=giphy.gif&ct=g)